ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```
@@ -0,0 +1,885 @@
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "source" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "formats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "topicName" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamSinkV1",
+      "required" : [ "@type", "properties", "source", "formats", "topicName" ]
+    },
+    "StreamSource" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamSourceV1" ],
+          "default" : "streamSourceV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "offsetReset" : {
+          "type" : "string",
+          "enum" : [ "EARLIEST", "LATEST" ]
+        },
+        "sourceName" : {
+          "type" : "string"
+        },
+        "alias" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamSourceV1",
+      "required" : [ "@type", "properties", "offsetReset", "sourceName", "alias" ]
+    },
+    "WindowedStreamSource" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "windowedStreamSourceV1" ],
+          "default" : "windowedStreamSourceV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "offsetReset" : {
+          "type" : "string",
+          "enum" : [ "EARLIEST", "LATEST" ]
+        },
+        "sourceName" : {
+          "type" : "string"
+        },
+        "alias" : {
+          "type" : "string"
+        }
+      },
+      "title" : "windowedStreamSourceV1",
+      "required" : [ "@type", "properties", "offsetReset", "sourceName", "alias" ]
+    },
+    "StreamStreamJoin" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamStreamJoinV1" ],
+          "default" : "streamStreamJoinV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "joinType" : {
+          "type" : "string",
+          "enum" : [ "INNER", "LEFT", "OUTER" ]
+        },
+        "leftFormats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "rightFormats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "left" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "right" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "before" : {
```

```suggestion
        "beforeDuration" : {
```

Or something to make it clearer what this is? Maybe:

```json
   "joinWindow": {
       "beforeMillis": {},
       "afterMillis": {}
   }
```